### PR TITLE
feat: add spatial navigation focusable view

### PR DESCRIPTION
**Pull request by remilry**
| iOS | Android | Web |
| ---  | ------- | ----- |
| SimulatorScreenRecording-AppleTV-2024-02-21at16 51 10-ezgif com-optimize| Feb-21-202416-53-43-ezgif com-resize|Feb-21-202416-56-51-ezgif com-resize|

### CODE REVIEW

File: packages/lib/src/spatial-navigation/components/FocusableView.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    | React.ReactElement
+    | ((props: { isFocused: boolean; isActive: boolean }) => React.ReactElement);
+};
+
+export const SpatialNavigationFocusableView = forwardRef<SpatialNavigationNodeRef, Props>(
+  ({ children, ...props }: Props, ref) => {
+    return (
+      <SpatialNavigationNode isFocusable {...props} ref={ref}>
+        {({ isFocused, isActive }) => (
+          <View>
```
Comment (pierpo) [inline packages/lib/src/spatial-navigation/components/FocusableView.tsx:17]: You should definitely pass an option `style` property ;)

File: packages/example/src/modules/program/view/ProgramNode.tsx [hunk 2]
```diff
@@ -15,9 +15,9 @@ type Props = {
 export const ProgramNode = forwardRef<SpatialNavigationNodeRef, Props>(
```
Comment (pierpo) [inline packages/example/src/modules/program/view/ProgramNode.tsx:15]: you can get rid of the forward ref I think 🤔

File: packages/lib/src/spatial-navigation/components/FocusableView.tsx (hunk 1 above)
Comment (remilry) [reply packages/lib/src/spatial-navigation/components/FocusableView.tsx:17]: True !

File: packages/example/src/modules/program/view/ProgramNode.tsx (hunk 2 above)
Comment (pierpo) [reply packages/example/src/modules/program/view/ProgramNode.tsx:15]: Oops indeed you're right!

**Review by pierpo — APPROVED** [review]

**Pull request merged**